lib/rabbit_feed/connection.rb

```diff
@@ -0,0 +1,22 @@
+module RabbitFeed
+  class Connection
+    include Singleton
+
+    def initialize
+      @connection = Bunny.new RabbitFeed.configuration.connection_options
+      @connection.start
+      @channel = @connection.create_channel
+      @mutex = Mutex.new
+    end
+
+    protected
+
+    attr_reader :channel
+
+    def thread_safe &block
```

Comment:
I don't like the name `thread_safe` here very much. Thread safe in general implies more than synchronizing access to a part of code (even if that is the way it is achieved here), and also I can't imagine why anyone would not use a thread safe version. I would prefer to call this method something else... not sure what... maybe something like `serialized` or `sequential`.. or nothing specific at all.. simply `execute`